modules/component-store/src/component-store.ts

```diff
@@ -225,46 +226,53 @@ export class ComponentStore<T extends object> implements OnDestroy {
     projector: (s: T) => Result,
     config?: SelectConfig
   ): Observable<Result>;
+  select<SelectorsObject extends Record<string, Observable<unknown>>>(
+    selectorsObject: SelectorsObject,
+    config?: SelectConfig
+  ): Observable<{
+    [K in keyof SelectorsObject]: ObservedValueOf<SelectorsObject[K]>;
+  }>;
   select<Selectors extends Observable<unknown>[], Result>(
-    ...args: [...selectors: Selectors, projector: Projector<Selectors, Result>]
+    ...selectorsWithProjector: [
+      ...selectors: Selectors,
+      projector: Projector<Selectors, Result>
+    ]
   ): Observable<Result>;
   select<Selectors extends Observable<unknown>[], Result>(
-    ...args: [
+    ...selectorsWithProjectorAndConfig: [
       ...selectors: Selectors,
       projector: Projector<Selectors, Result>,
       config: SelectConfig
     ]
   ): Observable<Result>;
   select<
-    Selectors extends Array<Observable<unknown> | SelectConfig | ProjectorFn>,
+    Selectors extends Array<
+      Observable<unknown> | SelectConfig | ProjectorFn | SelectorsObject
+    >,
     Result,
-    ProjectorFn extends (...a: unknown[]) => Result
+    ProjectorFn extends (...a: unknown[]) => Result,
+    SelectorsObject extends Record<string, Observable<unknown>>
   >(...args: Selectors): Observable<Result> {
-    const { observables, projector, config } = processSelectorArgs<
-      Selectors,
-      Result,
-      ProjectorFn
-    >(args);
-
-    let observable$: Observable<Result>;
-    // If there are no Observables to combine, then we'll just map the value.
-    if (observables.length === 0) {
-      observable$ = this.stateSubject$.pipe(
-        config.debounce ? debounceSync() : (source$) => source$,
-        map((state) => projector(state))
-      );
-    } else {
-      // If there are multiple arguments, then we're aggregating selectors, so we need
-      // to take the combineLatest of them before calling the map function.
-      observable$ = combineLatest(observables).pipe(
-        config.debounce ? debounceSync() : (source$) => source$,
-        map((projectorArgs) => projector(...projectorArgs))
+    const { observablesOrSelectorsObject, projector, config } =
+      processSelectorArgs<Selectors, Result, ProjectorFn, SelectorsObject>(
+        args
       );
-    }
 
-    return observable$.pipe(
+    const source$ = hasProjectFnOnly(observablesOrSelectorsObject, projector)
+      ? this.stateSubject$
+      : combineLatest(observablesOrSelectorsObject as any);
+
+    return source$.pipe(
+      config.debounce ? debounceSync() : noopOperator(),
+      (projector
+        ? map((projectorArgs) =>
+            Array.isArray(projectorArgs)
+              ? projector(...projectorArgs)
+              : projector(projectorArgs)
+          )
+        : noopOperator()) as () => Observable<Result>,
       distinctUntilChanged(),
-      shareReplay({
+      shareReplay<Result>({
```

Comment:
`Result` should be automatically inferred (after line 273)

```suggestion
      shareReplay({
```

Reply:
That's right.